<?xml version="1.0"?>
<launch>
    <include file="$(find-pkg-share forgescan_realsense)/launch/load_forgescan_realsense.launch.xml" />

    <!-- Input Arguments -->
    <arg name="sim_robot" default="True" description="[Bool] Robot Joint State Simulated or listen for /joint_states topic?" />
    <arg name="rviz_config" default="$(find-pkg-share forgescan_support)/config/visualize_description.rviz" description="[String] Path to RViz config file." />

    <!-- Simulate Joint State -->
    <group if="$(var sim_robot)" >
        <node pkg="joint_state_publisher_gui" exec="joint_state_publisher_gui" output="screen" />
    </group>

    <!-- Publish the robot state (tf transforms) -->
    <node pkg="robot_state_publisher" exec="robot_state_publisher" output="screen" >
        <param name="robot_description" value="$(var robot_description)" />
    </node>

<!--     User Interaction -->
    <node pkg="rviz2" exec="rviz2" args="-d $(var rviz_config)" >
        <param name="robot_description" value="$(var robot_description)" />
    </node>

    <node pkg="realsense2_camera" exec="realsense2_camera_node">
        <remap from="/camera/camera/depth/image_rect_raw" to="forgescan_realsense/camera_image"/>
    </node>

    <node pkg="forgescan_realsense" exec="realsense_handler"/>
    <node pkg="forgescan_realsense" exec="scan_manager"/>
    <node pkg="forgescan_realsense" exec="scan_image"/>
</launch>